R/stroke.R

```diff
@@ -150,6 +171,12 @@ best_link <- function(nodes, segments, links, angle_threshold = 0) {
                              nodes[end_node, , drop = FALSE],
                              nodes[linked_nodes, , drop = FALSE])
     best_link <- get_best_link(angles, linked_segs, angle_threshold_rad)
+
+    # if flow_mode, we choose the link on the same edge
+    if (flow_mode) {
+      best_link <- get_link_on_same_edge(linked_segs, edge_ids, edge_id)
+    }
```

Comment:
Nice way to find the continuation link on the same edge! But what happens if none of the linked segments falls on the same edge? If I understand it correctly,  `get_link_on_same_edge` would then return an empty array, the assignment on line 180 would be skipped, and the  `best_links` element would remain NA. 

I think this is not what should happen here: when we have no more continuation links on the same edge (i.e. when we reach the end of an edge) we actually have to look for the best-angle continuation of the segment. So I wonder whether  one should maybe swap the order of the last few lines? Something like:
1. Call `get_linked_segments` to find out which  continuation links we have;
2. If `flow_mode`, call `get_link_on_same_edge` to find out whether we have a candidate `best_link`
3. If `(lenght(best_link) == 0` (and if not `flow_mode` ),  call `get_linked_nodes` , `interior_angle`, and `get_best_link` to find out the best possible continuation of the segment. 
4.  Only if `(lenght(best_link) > 0` , proceed to the assignment, leave the best link as NA otherwise. 

What do you think? I have to say I haven't tried running these steps in this order, so I am not 100% this will work. Also I am curious if you have more insight by having actually run the tests for `flow_mode = TRUE` (which seems to pass!)..

Reply:
Correct, I fixed the order of the operations and the tests are passed.